src/Stormpath/Resource/Tenant.php

```diff
@@ -71,6 +72,18 @@ public function getDirectories(array $options = array())
         return $this->getResourceProperty(self::DIRECTORIES, Stormpath::DIRECTORY_LIST, $options);
     }
 
+    public function getCustomData(array $options = array())
+    {
+        $customData =  $this->getResourceProperty(self::CUSTOM_DATA, Stormpath::CUSTOM_DATA, $options);
+
+        if(!$customData) {
+            $customData = new CustomData();
+            $this->setProperty(self::CUSTOM_DATA, $customData);
+        }
+
+        return $customData;
+    }
```

Comment:
Will work on this.  There is an issue because a few of the resources use a different class it extends, causing alittle bit of redundancy.  Since I can not use traits, i will have to find another way